File: week_1/day-7-revision/range_sum.js
// range_sum.js

class RangeSum {
    constructor(nums) {
        this.prefix = new Array(nums.length + 1).fill(0);
        for (let i = 0; i < nums.length; i++) {
            this.prefix[i + 1] = this.prefix[i] + nums[i];
        }
    }

    query(l, r) {
        return this.prefix[r + 1] - this.prefix[l];
    }
}

// Example Input
const nums = [1, 2, 3, 4, 5];
const rs = new RangeSum(nums);

// Queries
console.log(rs.query(0, 2)); // Output: 6 (1+2+3)
console.log(rs.query(1, 3)); // Output: 9 (2+3+4)

File: week_1/day-7-revision/range_sum.py
# range_sum.py

class RangeSum:
    def __init__(self, nums):
        self.prefix = [0] * (len(nums) + 1)
        for i in range(len(nums)):
            self.prefix[i + 1] = self.prefix[i] + nums[i]

    def query(self, l, r):
        return self.prefix[r + 1] - self.prefix[l]

# Example Input
nums = [1, 2, 3, 4, 5]
rs = RangeSum(nums)

# Queries
print(rs.query(0, 2))  # Output: 6 (1+2+3)
print(rs.query(1, 3))  # Output: 9 (2+3+4)
